src/main/java/bio/terra/service/filedata/azure/util/BlobContainerCopyInfo.java

```diff
@@ -0,0 +1,57 @@
+package bio.terra.service.filedata.azure.util;
+
+import com.azure.core.util.polling.LongRunningOperationStatus;
+import com.azure.core.util.polling.PollResponse;
+import com.azure.storage.blob.models.BlobCopyInfo;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * This class encapsulates all poll responses for a given copy operation. From the responses the
+ * status of the operation is determined.
+ */
+public class BlobContainerCopyInfo {
+  private final List<PollResponse<BlobCopyInfo>> pollResponses;
+
+  public BlobContainerCopyInfo(List<PollResponse<BlobCopyInfo>> pollResponses) {
+    this.pollResponses = pollResponses;
+  }
+
+  /**
+   * Returns the status of the copy operation from the individual poll responses. The implementation
+   * follows the these rules in order: SUCCESSFULLY_COMPLETED: When all operations are completed
+   * successful. FAILED: When at least one operation has FAILED. USER_CANCELLED: When at least one
+   * operation has been cancelled. IN_PROGRESS: When at least one operation is in progress.
+   * NOT_STARTED: When at least one operation has not started.
+   *
+   * @return The status of the long running operation.
+   */
+  public LongRunningOperationStatus getCopyStatus() {
+
+    List<LongRunningOperationStatus> distinctStatuses =
+        pollResponses.stream().map(PollResponse::getStatus).distinct().collect(Collectors.toList());
```

Comment:
Given the way this is used, it would be easier to use a `Set<>`, which is always `distinct()` and supports a fast `contains()` operation. I would also use `var` here to avoid the long type declaration.
```suggestion
    var statuses = pollResponses.stream().map(PollResponse::getStatus).collect(Collectors.toSet());
```

Reply:
Fixed.

Reply:
Added in the last commit.